src/generators/static/README.md.hbs

```diff
@@ -0,0 +1,101 @@
+# {{ clientDescriptiveName }} library for JavaScript
+
+This package contains an isomorphic SDK (runs both in node.js and in browsers) for {{ clientDescriptiveName }}.
+
+{{ description }}
+
+{{#if packageSourceURL}}
+[Source code]({{ packageSourceURL }}) |
+{{/if}}
+{{#if packageNPMURL}}
+[Package (NPM)]({{ packageNPMURL }}) |
+{{/if}}
+{{#if apirefURL}}
+[API reference documentation]({{ apirefURL }}) |
+{{/if}}
+{{#if samplesURL}}
+[Samples]({{samplesURL}})
+{{/if}}
+
+## Getting started
+
+### Currently supported environments
+
+- Node.js version 8.x.x or higher
+- Browser JavaScript
+
+{{#if azure}}
+### Prerequisites
+
+- An [Azure subscription][azure_sub].
+{{/if}}
+
+### Install the `{{ clientPackageName }}` package
+
+Install the {{ clientDescriptiveName }} library for JavaScript with `npm`:
+
+```bash
+npm install {{ clientPackageName }}
+```
+
+### Create and authenticate a `{{ clientClassName}}`
+
+To create a client object to access the {{ serviceName }} API, you will need the `endpoint` of your {{ serviceName }} resource and a `credential`.{{#if azure}} The {{ clientDescriptiveName }} can use Azure Active Directory credentials to authenticate.
```

Comment:
Do all auto generated clients take "endpoint"?

Reply:
I believe so, @joheredi could we generate a client that does not take an endpoint?

Reply:
> can use Azure Active Directory credentials to authenticate

This should be behind the `addCredentials` check as well right?
And are we sure `addCredentials` is for indicating AAD support? Because we have "credentials" that are not AAD as well.

Reply:
Just looked at Track 1 mgmt plane package. The client constructor there does not require the endpoint.

Reply:
`addCredentials` means make the client constructor take the standard identit's credential type as input which implies that you do not have any fancy auth requirements. So I think it makes sense to make the AAD part of the readme behind that flag since it is part of the standard credential. If the package has custom auth requirements (e.g. API keys, conn strings, etc) then the author will most likely set that flag to false and implement these requirements manually in the convenience layer. At least this is what I saw so far.

Reply:
Ok I put that section behind `addCredentials` as well.

Reply:
I think we can merge this PR with the current behavior and we can revisit the endpoint issue later